packages/components/src/internal/watchEvent.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { PropertyValues } from 'lit';
+import type SynergyElement from './synergy-element.js';
+
+type Constructor<T = object> = new (...args: any[]) => T;
+
+interface DecoratorOptions {
+  /**
+   * If true, will only start watching after the initial updated call
+   */
+  waitUntilFirstUpdated?: boolean;
+}
+
+/**
+ * Class decorator that automatically emits a `syn-attributes-changed` event
+ * when at least one of the provided attributes have changed.
+ *
+ * @example
+ * ```typescript
+ * // Will fire a `syn-attributes-changed` event when `prop1` or `prop2` change,
+ * // but only after the first `updated` call
+ * emitEventForPropertyUpdates(['prop1', 'prop2'], {
```

Comment:
The `@` is missing in the example

```suggestion
 * @emitEventForPropertyUpdates(['prop1', 'prop2'], {
```

Reply:
I adjusted the documentation for this so it works without decorator syntax (which is the default as decorators are not yet standard). :)